RFCs/0011-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,772 @@
+
+```SQL
+-- Existing Item is:
+--  {HK: 1, RK: 1, myAttr: 12 }
+-- Outcome is:
+--  {HK: 1, RK: 1, thirdAttr: "World"}
+
+INSERT into Customers
+VALUES <<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello"}>>
+ON CONFLICT
+    DO REPLACE {HK: 1, RK: 1, thirdAttr: "World"};
+```
+
+##### Example 4.2.11
+
+The following example SHOULD* lead to replacing the existing item with the item specified by `REPLACE` as a new item:
+
+```SQL
+-- Existing items are:
+--  {HK: 1, RK: 1, myAttr: 12 }, 
+--  {HK: 1, RK: 2, myAttr: 12 }
+-- Outcome:
+--  {HK: 1, RK: 3, thirdAttr: “World”}, 
+--  {HK: 1, RK: 2, myAttr: 12 }
+
+INSERT into Customers
+VALUES <<{HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello" }>>
+ON CONFLICT
+    DO REPLACE {HK: 1, RK: 3, thirdAttr: "World"};
+```
+
+** SHOULD, because for a distributed database system, implementing this logic may lead to data inconsistency. This is because there is no guarantee that the INSERT procedure that generates the conflict executes on the same database host where the conflict is going to get resolved.
+
+##### Example 4.2.12
+
+The following example MUST lead to a `SemanticError` if there is an existing item that conflicts with the item specified by replace:
+
+```SQL
+-- Existing items is:
+--  {HK: 1, RK: 1, myAttr: 12 },
+--  {HK: 1, RK: 2, myAttr: 12 }
+-- Outcome is SemanticError
+
+INSERT into Customers
+VALUES <<{ HK: 1, RK: 1, myAttr: 12, anotherAttr: "Hello" }>>
+ON CONFLICT
+    DO REPLACE {HK: 1, RK: 2, thirdAttr: "World"};
+```
+
+## 5. Other Examples
+
+To insert multiple rows as bag value using the multi–row `VALUES` syntax:
+
+```SQL
+INSERT INTO films
+VALUES <<{code: 'B6717', title: 'Tampopo', did: 110},
+    {code: 'HG120', title: 'The Dinner Game', did: 140}>>;
+```
+
+Insert or update new distributors as appropriate. Assumes a unique index has been defined that constrains values appearing in the `did` attribute. Note that the special `excluded` table is used to reference values originally proposed for insertion:
+
+```SQL
+INSERT INTO distributors (did, dname)
+VALUES (5, 'Gizmo Transglobal'),
+       (6, 'Associated Computing, Inc') ON CONFLICT (did) 
+       DO
+UPDATE SET dname = EXCLUDED.dname;
+```
+
+Insert or update new distributors values as bag value. Assumes the database engine implementation can infer the primary key for `<conflict target>`. Note that the special `EXCLUDED` table is used to reference values originally proposed for insertion:
+
+```SQL
+INSERT INTO distributors
+VALUES <<{did: 5, dname: 'Gizmo Transglobal'},
+    {did: 6, dname: 'Associated Computing, Inc'}>>
+ON CONFLICT
+    DO
+UPDATE SET dname = EXCLUDED.dname;
+```
+
+Insert distributors values as bag value. In case of a conflict on unique constraint updates the values as stated in the insert. Assumes the database engine implementation can infer the unique index as . Note that the special `excluded` table is used to reference values originally proposed for insertion:
```

Comment:
Just spotted this out of the corner of my eye:

> ... implementation can infer the unique index as .

Incomplete sentence?

Reply:
Yes, fixed in the new commit.